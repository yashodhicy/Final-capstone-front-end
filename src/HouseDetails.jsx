import React, { useEffect, useState } from "react"
import { useDispatch, useSelector } from "react-redux";
import { useNavigate, useParams } from "react-router-dom";

const HouseDetails = () => {
  const dispatch = useDispatch();
  const { houseId } = useParams();
  const navigate = useNavigate();
  const userId =1;

  const houses = useSelector((state) => state.Houses).houses;
  const houseIdNumber = parseInt(houseId, 10);
  const house = houses.find((house) => houseIdNumber === house.id);
  console.log(house);

  const [deleteMessage, setDeleteMessage] = useState("");
  
  const handleDelete = () => {
      dispatch(deleteHouseThunk(houseId)).then((action) => {
        const message = 'Successfully Deleted';
        navigate(`/?message=${encodeURIComponent(message)}`);
    })
    .catch((error) => {
      const message = 'Error: Unable to delete';
    });
  };

  if (house) {
  return (
    <div className="house-details">
      <div className="details">
        <img src={house.image} alt="House" />
        <h2>{house.name}</h2>
        <p>Location: {house.location}</p>
        <p>Area : {house.area}</p>
        <p>Number of Rooms: {house.number_of_rooms}</p>
        <p>Price per day: {house.price} $</p>
        <div>
          <button
            type="button"
            className="reserve"
            onClick={() => console.log("reserve")}
          >
            Reserve
          </button>
        </div> 
      </div>
      {house.user_id === userId &&
      (
      <div className="delete">
        <button type="button" className="delete" onClick={handleDelete}>
          Delete House
        </button>
      </div>
      )}
    </div>
  );
  } 
  else {
    return <div>Loading...</div>;
  }
};

export default HouseDetails;
