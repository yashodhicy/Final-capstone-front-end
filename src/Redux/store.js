import { configureStore } from '@reduxjs/toolkit';
import HouseSlice from './HouseSlice';
import reducer from './reservation/reducer';


const store = configureStore({
  reducer: {
    Houses: HouseSlice,
    reservations: reducer
  },
});

export default store;
