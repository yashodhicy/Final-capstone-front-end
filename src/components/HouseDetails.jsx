import "./componentsCss/housedetails.css";
import 'font-awesome/css/font-awesome.min.css';
import { useSelector } from "react-redux";
import { useNavigate, useParams } from "react-router-dom";
import { Container, Row, Col, Image, Table} from "react-bootstrap";




const HouseDetails = () => {
  const { houseId } = useParams();
  const navigate = useNavigate();

  const houses = useSelector((state) => state.Houses).houses;
  const houseIdNumber = parseInt(houseId, 10);
  const house = houses.find((house) => houseIdNumber === house.id);
  
  const handleReserve = () => {
    navigate("/reserve");
  };

  if (house) {
    return (
      <div className="house-details">
        <Container className="details">
          <Row>
            <Col md={8} className="house-imgc">
              <Image src={house.image} className="house-image m-2" />
              
            </Col>
            <Col className="info" md={4} >
              <h2>{house.name.toUpperCase()}</h2>
              <Table striped hover className="table">
                <tbody>
                  <tr>
                    <td>Location</td>
                    <td>{house.location}</td>
                  </tr>
                  <tr>
                    <td>Area</td>
                    <td>{house.area}</td>
                  </tr>
                  <tr>
                    <td>Number of Rooms</td>
                    <td>{house.number_of_rooms}</td>
                  </tr>
                  <tr>
                    <td>Price per day</td>
                    <td>{house.price} $</td>
                  </tr>
                </tbody>
              </Table>

              <div>
              <button size="lg"
                  className="reserve-btn"
                  onClick={handleReserve}
                >
                  <i class="fa fa-home fa-2x " aria-hidden="true"></i>

                  RESERVE

                  <i class="fa fa-chevron-circle-right fa-2x" aria-hidden="true"></i>
                </button>
              </div>
            </Col>
          </Row>
          <button className="prev" type="button" onClick={()=> navigate(-1)} >
            <i class="fa fa-caret-left fa-2x" aria-hidden="true"></i>
        </button>
          
        </Container>
        
      </div>
    );
  } else {
    return <div>Loading...</div>;
  }
};

export default HouseDetails;
