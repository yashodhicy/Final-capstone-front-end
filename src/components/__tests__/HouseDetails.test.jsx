/* eslint-env jest */
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../Redux/store';
import HouseDetails from '../HouseDetails';

describe('Test House Details Component', () => {
    test ('it renders correctly', () => {
        const tree = renderer
        .create(
            <Provider store={store}>
                <Router>
                    <HouseDetails />
                </Router>
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
});
