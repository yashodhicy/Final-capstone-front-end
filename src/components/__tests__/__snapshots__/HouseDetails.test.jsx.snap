// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test House Details Component it renders correctly 1`] = `
<div>
  Loading...
</div>
`;
